core/api/src/main/java/org/trellisldp/api/MutableDataService.java

```diff
@@ -34,10 +34,12 @@
      * @param session the session context for this operation
      * @param ixnModel the LDP interaction model for this resource
      * @param container an LDP container for this resource, {@code null} for none
+     * @param binary metadata for a binary resource, relevant only for ldp:NonRDFSource items: {@code null} for none
```

Comment:
I know what you mean, but "metadata for a binary resource" seems easily confused with "descriptive triples" (e.g. technical metadata, descriptive metadata). Maybe something as short as `a binary resource representation for {@code ldp:NonRDFSource} items, {@code null} for none`? I know that there is a real distinction between the bitstream, the triples, the info in a `Binary`-- just trying to be clear without being absurdly verbose.

Reply:
Also, this may not be the PR to do it in, but I think we should start shuffling the `null`able params to the end of the sig? That's my usual style, because they're more "optional"-ish...

Reply:
I'll rework the javadoc.

As for the param order, I agree. In fact, I was thinking that moving toward a `Builder` pattern which can more easily handle these "optional-ish" params. I'd like to save that for a separate PR, since this one is really focused on how the `Binary` object gets passed around these interfaces.

Reply:
A builder could be really nice, but I would want to watch out for the creation of a bazillion short-lived objects. I know that's not as big a problem as once it was, but still. Definitely not for this PR.

Reply:
Sounds good -- I'll move the param order around as part of a separate effort. Then, if a builder still makes sense, it could be added even further down the road.

Reply:
👍